rust/processor/src/models/ans_models/ans_lookup.rs

```diff
@@ -0,0 +1,776 @@
+        let value = &table_item_data.value;
+        let primary_name_record: NameRecordKeyV1 =
+            serde_json::from_str(value).unwrap_or_else(|e| {
+                panic!(
+                    "version {} failed! failed to parse type {}, data {:?}. Error: {:?}",
+                    txn_version, value_type, value, e
+                )
+            });
+
+        // Case: when a primary name points to a new address, we need to set the previous primary name record's to non-primary.
+        // Lookup the previous primary name record by target address. It may not exist.
+        if let Some(mut previous_primary_record) =
+            Self::lookup_name_record_by_primary_address(address_to_primary_name_record, key, conn)
+        {
+            // Update the old primary name record
+            previous_primary_record.is_primary = false;
+            previous_primary_record.last_transaction_version = txn_version;
+            new_current_ans_lookups.insert(
+                (
+                    previous_primary_record.domain.clone(),
+                    previous_primary_record.subdomain.clone(),
+                ),
+                previous_primary_record.clone(),
+            );
+            new_ans_lookups.insert(
+                (
+                    txn_version,
+                    previous_primary_record.domain.clone(),
+                    previous_primary_record.subdomain.clone(),
+                ),
+                AnsLookup {
+                    transaction_version: txn_version,
+                    domain: previous_primary_record.domain,
+                    subdomain: previous_primary_record.subdomain,
+                    registered_address: previous_primary_record.registered_address,
+                    expiration_timestamp: previous_primary_record.expiration_timestamp,
+                    token_name: previous_primary_record.token_name,
+                    is_primary: false,
+                    is_deleted: previous_primary_record.is_deleted,
+                    transaction_timestamp: txn_timestamp,
+                },
+            );
+        }
+
+        // If the name record doesn't exist yet, create it and insert it to current_ans_lookups.
+        // This is possible if the primary name record wsc happens before the name record wsc.
+        new_current_ans_lookups.insert(
+            (
+                primary_name_record.get_domain().clone(),
+                primary_name_record.get_subdomain().clone(),
+            ),
+            Self {
+                domain: primary_name_record.get_domain().clone(),
+                subdomain: primary_name_record.get_subdomain().clone(),
+                registered_address: Some(String::from(key)),
+                last_transaction_version: txn_version,
+                // Default to zero
+                expiration_timestamp: chrono::NaiveDateTime::from_timestamp_opt(0, 0)
+                    .unwrap_or_default(),
```

Comment:
how is this possible? I think that we should just unwrap? 

Also I'm confused why you're not using the actual expiration timestamp from the table item.

Reply:
We can default to null